clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/agent/KubernetesServerGroupCachingAgent.groovy

```diff
@@ -347,14 +347,14 @@ class KubernetesServerGroupCachingAgent extends KubernetesCachingAgent implement
           }
         }
 
-        loadBalancerKeys.forEach { loadBalancerKey ->
+        if (loadBalancerKeys != null) loadBalancerKeys.forEach { loadBalancerKey ->
           cachedLoadBalancers[loadBalancerKey].with {
             relationships[Keys.Namespace.SERVER_GROUPS.ns].add(serverGroupKey)
             relationships[Keys.Namespace.INSTANCES.ns].addAll(instanceKeys)
           }
         }
 
-        cachedServerGroups[serverGroupKey].with {
+        if (cachedServerGroups != null) cachedServerGroups[serverGroupKey].with {
```

Comment:
going to not change anything here since the bug seems to be fixed w/o it. will change if I encounter a bug in the future with this line.